src/wallet.js

```diff
@@ -500,14 +500,36 @@ const wallet = {
    * @inner
    */
   getAddressToUse() {
-    const address = storage.getItem('wallet:address');
+    const address = this.getCurrentAddress();
     // Updating address because the last one was used
+    this.updateCurrentAddress();
+    return address;
+  },
+
+  /**
+   * Get current address
+   *
+   * @return {string} address
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  getCurrentAddress() {
+    return storage.getItem('wallet:address');
+  },
+
+  /**
+   * Update current address, if has not yet reached the GAP_LIMIT
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  updateCurrentAddress() {
```

Comment:
I don't like the method name. Maybe `nextAddress`? Or `getNextAddress`?

Reply:
I changed to `getNextAddress `